app/models/samples/json/model_with_humidty_setpoints.json

```diff
@@ -0,0 +1,1588 @@
+                },
+                {
+                    "type": "EnergyMaterial",
+                    "name": "Generic Gypsum Board",
+                    "roughness": "MediumSmooth",
+                    "thickness": 0.0127,
+                    "conductivity": 0.16,
+                    "density": 800.0,
+                    "specific_heat": 1090.0,
+                    "thermal_absorptance": 0.9,
+                    "solar_absorptance": 0.5,
+                    "visible_absorptance": 0.5
+                },
+                {
+                    "type": "EnergyMaterial",
+                    "name": "Generic Wall Air Gap",
+                    "roughness": "Smooth",
+                    "thickness": 0.1,
+                    "conductivity": 0.667,
+                    "density": 1.28,
+                    "specific_heat": 1000.0,
+                    "thermal_absorptance": 0.9,
+                    "solar_absorptance": 0.7,
+                    "visible_absorptance": 0.7
+                },
+                {
+                    "type": "EnergyWindowMaterialGlazing",
+                    "name": "Generic Clear Glass",
+                    "thickness": 0.006,
+                    "solar_transmittance": 0.77,
+                    "solar_reflectance": 0.07,
+                    "solar_reflectance_back": 0.07,
+                    "visible_transmittance": 0.88,
+                    "visible_reflectance": 0.08,
+                    "visible_reflectance_back": 0.08,
+                    "infrared_transmittance": 0.0,
+                    "emissivity": 0.84,
+                    "emissivity_back": 0.84,
+                    "conductivity": 1.0,
+                    "dirt_correction": 1.0,
+                    "solar_diffusing": false
+                },
+                {
+                    "type": "EnergyMaterial",
+                    "name": "Generic HW Concrete",
+                    "roughness": "MediumRough",
+                    "thickness": 0.2,
+                    "conductivity": 1.95,
+                    "density": 2240.0,
+                    "specific_heat": 900.0,
+                    "thermal_absorptance": 0.9,
+                    "solar_absorptance": 0.8,
+                    "visible_absorptance": 0.8
+                },
+                {
+                    "type": "EnergyMaterial",
+                    "name": "Generic 25mm Wood",
+                    "roughness": "MediumSmooth",
+                    "thickness": 0.0254,
+                    "conductivity": 0.15,
+                    "density": 608.0,
+                    "specific_heat": 1630.0,
+                    "thermal_absorptance": 0.9,
+                    "solar_absorptance": 0.5,
+                    "visible_absorptance": 0.5
+                }
+            ],
+            "program_types": [
+                {
+                    "type": "ProgramTypeAbridged",
+                    "name": "2013::Hospital::ICU_PatRm",
```

Comment:
@chriswmackey I had not added `:` in the valid regex. Should I make that change?

Reply:
The filename is misspelled.

Reply:
Yes, I think we should change all of the `name` validators in the schema sooner rather than later.  You can leave the name validation of the geometry objects (Room, Face, Shade, Aperture, Door) as they are right now.  However, all of the energy simulation object names (eg. Materials, Constructions, Schedules, ProgramTypes, etc.) are using this code on the Python end to ensure strings are valid for EnergyPlus:
https://github.com/ladybug-tools/honeybee-core/blob/master/honeybee/typing.py#L47-L64

Really, the only restrictions are:
1. All characters must be ASCII.
2. The name can't include one of these characters: `,`, `;`, `!`, `\n`, `\t` (note that the second one is a semicolon and not a colon)
3. The name must be between 1-100 characters long.